pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -432,20 +432,28 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 		out.ResponseHeadersToRemove = responseHeadersToRemove
 
 		if in.Mirror != nil {
+			cluster := GetDestinationCluster(in.Mirror, serviceRegistry[host.Name(in.Mirror.Host)], port)
 			var percent uint32 = 100
+			var percentage *networking.Percent = &networking.Percent{Value: 100}
 			if in.MirrorPercent != nil {
 				percent = in.MirrorPercent.GetValue()
 			}
-
-			if percent > 0 {
-				n := GetDestinationCluster(in.Mirror, serviceRegistry[host.Name(in.Mirror.Host)], port)
+			if in.MirrorPercentage != nil {
+				percentage = in.MirrorPercentage
+			}
+			switch {
+			case percent > 0:
 				action.RequestMirrorPolicy = &route.RouteAction_RequestMirrorPolicy{
-					Cluster: n,
+					Cluster: cluster,
 					RuntimeFraction: &core.RuntimeFractionalPercent{
-						DefaultValue: &xdstype.FractionalPercent{
-							Numerator:   percent,
-							Denominator: xdstype.FractionalPercent_HUNDRED,
-						},
+						DefaultValue: translateIntegerToFractionalPercent((int32(percent))),
+					},
+				}
+			case percentage.GetValue() > 0:
```

Comment:
I think these need to be inverted (percentage first and then percent) so that the new one is alwauys given precedence over the old.

Reply:
ok. Will change